### ISSUE TITLE
Schema dump fails in flutter with drift_dev 2.19

### ISSUE THREAD

**Issue by fmorato** [simolus3/moor#3097]
**Describe the bug**

I'm just learning to use drift, and looking at how to dump schemas, but it's failing. 
Is this a bug, or how am I supposed to use it?

This is what I did

```
$ git clone https://github.com/simolus3/drift.git
$ cd drift/examples/app
```
<details><summary>dart pub get</summary>
<p>

```
$ dart pub get
Resolving dependencies... (1.2s)
Downloading packages... (4.1s)
+ _fe_analyzer_shared 67.0.0 (73.0.0 available)
+ analyzer 6.4.1 (6.8.0 available)
+ analyzer_plugin 0.11.3
+ archive 3.6.1
+ args 2.5.0
+ async 2.11.0
+ bazel_worker 1.1.1
+ boolean_selector 2.1.1
+ build 2.4.1
+ build_config 1.1.1
+ build_daemon 4.0.2
+ build_modules 5.0.8 (5.0.9 available)
+ build_resolvers 2.4.2
+ build_runner 2.4.11
+ build_runner_core 7.3.1
+ build_web_compilers 4.0.10 (4.0.11 available)
+ built_collection 5.1.1
+ built_value 8.9.2
+ characters 1.3.0
+ charcode 1.3.1
+ checked_yaml 2.0.3
+ cli_util 0.4.1
+ clock 1.1.1
+ code_builder 4.10.0
+ collection 1.18.0 (1.19.0 available)
+ convert 3.1.1
+ cross_file 0.3.4+1
+ crypto 3.0.3
+ dart_style 2.3.6
+ drift 2.19.1
+ drift_dev 2.19.0
+ drift_flutter 0.1.0
+ fake_async 1.3.1
+ ffi 2.1.2
+ file 7.0.0
+ file_picker 8.0.6
+ fixnum 1.1.0
+ flutter 0.0.0 from sdk flutter
+ flutter_colorpicker 1.1.0
+ flutter_lints 4.0.0
+ flutter_plugin_android_lifecycle 2.0.20
+ flutter_riverpod 2.5.1
+ flutter_test 0.0.0 from sdk flutter
+ flutter_web_plugins 0.0.0 from sdk flutter
+ frontend_server_client 4.0.0
+ glob 2.1.2
+ go_router 14.2.1
+ graphs 2.3.1
+ http_multi_server 3.2.1
+ http_parser 4.0.2 (4.1.0 available)
+ intl 0.19.0
+ io 1.0.4
+ js 0.7.1
+ json_annotation 4.9.0
+ leak_tracker 10.0.4 (10.0.5 available)
+ leak_tracker_flutter_testing 3.0.3 (3.0.5 available)
+ leak_tracker_testing 3.0.1
+ lints 4.0.0
+ logging 1.2.0
+ matcher 0.12.16+1
+ material_color_utilities 0.8.0 (0.12.0 available)
+ meta 1.12.0 (1.15.0 available)
+ mime 1.0.5
+ package_config 2.1.0
+ path 1.9.0
+ path_provider 2.1.3
+ path_provider_android 2.2.7
+ path_provider_foundation 2.4.0
+ path_provider_linux 2.2.1
+ path_provider_platform_interface 2.1.2
+ path_provider_windows 2.3.0
+ platform 3.1.5
+ plugin_platform_interface 2.1.8
+ pool 1.5.1
+ protobuf 3.1.0
+ pub_semver 2.1.4
+ pubspec_parse 1.3.0
+ recase 4.1.0
+ riverpod 2.5.1
+ scratch_space 1.0.2
+ shelf 1.4.1 (1.4.2 available)
+ shelf_web_socket 2.0.0
+ sky_engine 0.0.99 from sdk flutter
+ source_gen 1.5.0
+ source_maps 0.10.12
+ source_span 1.10.0
+ sqlite3 2.4.4
+ sqlite3_flutter_libs 0.5.24
+ sqlparser 0.37.0
+ stack_trace 1.11.1
+ state_notifier 1.0.0
+ stream_channel 2.1.2
+ stream_transform 2.1.0
+ string_scanner 1.2.0 (1.3.0 available)
+ term_glyph 1.2.1
+ test_api 0.7.0 (0.7.3 available)
+ timing 1.0.1
+ typed_data 1.3.2
+ vector_math 2.1.4
+ vm_service 14.2.1 (14.2.4 available)
+ watcher 1.1.0
+ web 0.5.1 (1.0.0 available)
+ web_socket 0.1.6
+ web_socket_channel 3.0.1
+ win32 5.5.1
+ xdg_directories 1.0.4
+ yaml 3.1.2
Changed 107 dependencies!
15 packages have newer versions incompatible with dependency constraints.
Try `dart pub outdated` for more information.

```
</p>
</details>

```
$ dart run drift_dev schema dump lib/database/database.dart drift_schemas/
Building package executable... (2.2s)
Failed to build drift_dev:drift_dev:
../../../../.pub-cache/hosted/pub.dev/drift_dev-2.19.0/lib/src/services/schema/schema_files.dart:174:76: Error: Required named parameter 'withNullability' must be provided.
          'dart_type_name': column.typeConverter!.dartType.getDisplayString(),
                                                                           ^
Failed to build drift_dev:drift_dev:
../../../../.pub-cache/hosted/pub.dev/drift_dev-2.19.0/lib/src/services/schema/schema_files.dart:174:76: Error: Required named parameter 'withNullability' must be provided.
          'dart_type_name': column.typeConverter!.dartType.getDisplayString(),
                                                                           ^
```

<details><summary>flutter --version</summary>
<p>

```
$ flutter --version
Flutter 3.22.3 • channel stable • https://github.com/flutter/flutter.git
Framework • revision b0850beeb2 (29 hours ago) • 2024-07-16 21:43:41 -0700
Engine • revision 235db911ba
Tools • Dart 3.4.4 • DevTools 2.34.3
```

</p>
</details> 

<details><summary>flutter doctor</summary>
<p>

```
$ flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.22.3, on openSUSE Tumbleweed-Slowroll 6.9.7-1-default, locale
    en_US.UTF-8)
[✓] Android toolchain - develop for Android devices (Android SDK version 35.0.0)
[✓] Chrome - develop for the web
[✓] Linux toolchain - develop for Linux desktop
[✓] Android Studio (version 2024.1)
[✓] VS Code (version 1.89.1)
[✓] Connected device (2 available)
[✓] Network resources

• No issues found!
```

</p>
</details>

**Comment by simolus3** [comment]
Thanks for the report! This is indeed a bug in `drift_dev`. The reason it hasn't been caught is that later `analyzer` versions (that are not yet supported by the stable Flutter channel) no longer require the `withNullability` parameter (in fact, it's deprecated).
As `drift_dev` doesn't depend on Flutter, it gets the latest `analyzer` and I've forgotten that `withNullability` is required for older versions. In Flutter projects, you get an older analyzer and parts of `drift_dev` then fail to compile.

I have fixed this in 088bec149455adbaf1e084c0f4c815cdc10e197c and will issue a patch release soon.

**Comment by simolus3** [comment]
Fixed in `drift_dev` version 2.19.1.
